src/api/metadata.ts

```diff
@@ -6,39 +6,22 @@ import { authMiddleware, UseSuspenseQueryOptions } from './utils';
 
 const client = createClient<paths>({ baseUrl: config.apiEndpoint.metadata });
 client.use(authMiddleware);
-
-const metadataFullSubjectPath = '/api/v1/subject/full/';
-export function useMetadataFullSubjectModel({
-  params,
-  reactQuery,
-}: UseSuspenseQueryOptions<paths[typeof metadataFullSubjectPath]['get']>) {
-  return useSuspenseQuery({
-    ...reactQuery,
-    queryKey: [metadataFullSubjectPath, params],
-    queryFn: async ({ signal }) => {
-      const { data } = await client.GET(metadataFullSubjectPath, {
-        params,
-        signal, // allows React Query to cancel request
-      });
-      return data;
-    },
-  });
+function createMeatdataFetchingHook<K extends keyof paths>(path: K) {
+  return function ({ params, reactQuery }: UseSuspenseQueryOptions<paths[typeof path]['get']>) {
+    return useSuspenseQuery({
+      ...reactQuery,
+      queryKey: [path, params],
+      queryFn: async ({ signal }) => {
+        const { data } = await client.GET(path, {
+          params,
+          signal,
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } as any);
+        return data;
+      },
+    });
+  };
 }
 
-const metadataFullLibraryPath = '/api/v1/library/full/';
-export function useMetadataFullLibraryModel({
-  params,
-  reactQuery,
-}: UseSuspenseQueryOptions<paths[typeof metadataFullLibraryPath]['get']>) {
-  return useSuspenseQuery({
-    ...reactQuery,
-    queryKey: [metadataFullLibraryPath, params],
-    queryFn: async ({ signal }) => {
-      const { data } = await client.GET(metadataFullLibraryPath, {
-        params,
-        signal, // allows React Query to cancel request
-      });
-      return data;
-    },
-  });
-}
+export const useMetadataFullSubjectModel = createMeatdataFetchingHook('/api/v1/subject/full/');
```

Comment:
Hmmm, I like the idea of this. But it seems that the data returned here resolved to `any` which kinda defeat the purpose of having the types based on OpenAPI schema. Perhaps this might be just fixing the type syntax.